### PR TITLE
Add `MUXSDKCustomerData` monitor methods

### PR DESCRIPTION
**Pull request by nidhik**

### CODE REVIEW

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.h [hunk 1]
```diff
@@ -46,6 +46,17 @@ FOUNDATION_EXPORT
 - (_Null_unspecified instancetype)init NS_UNAVAILABLE;
 + (_Null_unspecified instancetype)new NS_UNAVAILABLE;
 
+/*!
+ @method      monitorAVPlayerViewController:withPlayerName:customerData:
+ @abstract    Starts to monitor a given AVPlayerViewController.
+ @param       player An AVPlayerViewController to monitor
+ @param       name A name for this instance of the player
+ @param       customerData A MUXSDKCustomerrData object with player, video, and view metadata
```
Comment (bmulligan-kg) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.h:54]: is the idea that MUXSDKCustomerData is going to contain a MUXCustomerPlayerData MUXCustomerVideoData and MUXCustomerViewData?

Comment (bmulligan-kg) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.h:54]: Typo `MUXSDKCustomerrData`

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.h [hunk 2]
```diff
@@ -107,6 +118,17 @@ FOUNDATION_EXPORT
  */
 + (void)updateAVPlayerViewController:(nonnull AVPlayerViewController *)player withPlayerName:(nonnull NSString *)name;
 
+/*!
+ @method      monitorAVPlayerLayer:withPlayerName:customerData:
+ @abstract    Starts to monitor a given AVPlayerLayer.
+ @param       player An AVPlayerLayer to monitor
+ @param       name A name for this instance of the player
+ @param       customerData A MUXSDKCustomerPlayerData object with player, video, and view metadata
```
Comment (bmulligan-kg) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.h:126]: should be `MUXSDKCustomerData`

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.h [hunk 3]
```diff
@@ -229,6 +251,15 @@ FOUNDATION_EXPORT
  */
 + (void)programChangeForPlayer:(nonnull NSString *)name withVideoData:(nullable MUXSDKCustomerVideoData *)videoData;
 
+/*!
+ @method      setCustomerData:forPlayer:
+ @abstract    allows customerData to be set or updated for the player
+ @param       name The name of the player to update
+ @param       customerData A MUXSDKCustomerPlayerData object with player, video, and view metadata
```
Comment (bmulligan-kg) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.h:258]: same here `MUXSDKCustomerData`

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.h (hunk 1 above)
Comment (nidhik) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.h:54]: Exactly

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m [hunk 4]
```diff
@@ -127,6 +137,53 @@ + (void)initSDK {
...
+                                  customerData:customerData
+                        automaticErrorTracking:true];
+}
+
++ (MUXSDKPlayerBinding *_Nullable)monitorAVPlayerViewController:(nonnull AVPlayerViewController *)player
+                                                 withPlayerName:(nonnull NSString *)name
+                                                   customerData:(nonnull MUXSDKCustomerData *)customerData
+                                         automaticErrorTracking:(BOOL)automaticErrorTracking {
+
+    [self initSDK:[customerData customerViewerData]];
```
Comment (bmulligan-kg) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.m:154]: note that customerViewer data is included and overwritten with _every_ monitorAVPlayer* call

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m [hunk 5]
```diff
@@ -383,13 +451,15 @@ + (void)setAutomaticVideoChange:(NSString *)name  enabled:(Boolean)enabled {
...
-
-+ (void)updateCustomerDataForPlayer:(nonnull NSString *)name withPlayerData:(nullable MUXSDKCustomerPlayerData *)playerData withVideoData:(nullable MUXSDKCustomerVideoData *)videoData viewData: (nullable MUXSDKCustomerViewData *) viewData {
++ (void)setCustomerData:(nullable MUXSDKCustomerData *)customerData forPlayer:(nonnull NSString *)name {
     MUXSDKPlayerBinding *player = [_viewControllers valueForKey:name];
     if (!player) return;
+
+    MUXSDKCustomerPlayerData *playerData = [customerData customerPlayerData];
+    MUXSDKCustomerViewData *viewData = [customerData customerViewData];
+    MUXSDKCustomerVideoData *videoData = [customerData customerVideoData];
+    _customerViewerData = [customerData customerViewerData]; // call initSDK?
```
Comment (bmulligan-kg) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.m:461]: this should _probably_ call initSDK again, or any changes to customerViewerData don't get packaged into an event until something else calls initSDK

Comment (nidhik) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:461]: alternatively, this could call out to `dispatchGlobalDataEvent` right?

Comment (nidhik) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:461]: Perhaps this piece in `initSDK` specifically is worth putting into a separate method so we can do that
```
MUXSDKViewerData *viewerData = [[MUXSDKViewerData alloc] init];
    NSString *bundleId = [[NSBundle mainBundle] bundleIdentifier];
    if (bundleId) {
        [viewerData setViewerApplicationName:bundleId];

    NSString *viewerApplicationName = [_customerViewerData viewerApplicationName];
    if (viewerApplicationName == nil) {
        viewerApplicationName = [[NSBundle mainBundle] bundleIdentifier];
    }
    [viewerData setViewerApplicationName:viewerApplicationName];
 ```

Comment (bmulligan-kg) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:461]: yes that is an option as well

**Comment by nidhik** [comment]
closing in favor of https://github.com/muxinc/mux-stats-sdk-avplayer/pull/110

**Pull request closed without merging**